horovod/spark/common/store.py

```diff
@@ -281,6 +295,17 @@ def local_run_path():
                 yield tmpdir
         return local_run_path
 
+    def get_local_run_dir_fn(self, run_id):
+        @contextlib.contextmanager
+        def local_run_path():
+            dirpath = os.path.join(tempfile._get_default_tempdir(),self.get_checkpoint_dir)
```

Comment:
this does not call into `get_checkpoint_dir` but uses the method as the argument, causing:

```
Tue Nov 23 11:59:00 2021[1,1]<stderr>:  File "/usr/local/lib/python3.8/dist-packages/horovod/spark/common/store.py", line 301, in local_run_path
Tue Nov 23 11:59:00 2021[1,1]<stderr>:    dirpath = os.path.join(tempfile._get_default_tempdir(),self.get_checkpoint_dir)
Tue Nov 23 11:59:00 2021[1,1]<stderr>:  File "/usr/lib/python3.8/posixpath.py", line 90, in join
Tue Nov 23 11:59:00 2021[1,1]<stderr>:    genericpath._check_arg_types('join', a, *p)
Tue Nov 23 11:59:00 2021[1,1]<stderr>:  File "/usr/lib/python3.8/genericpath.py", line 152, in _check_arg_types
Tue Nov 23 11:59:00 2021[1,1]<stderr>:    raise TypeError('%s() argument must be str or bytes, not %r' %
Tue Nov 23 11:59:00 2021[1,1]<stderr>:TypeError: join() argument must be str or bytes, not 'method'
```
https://github.com/horovod/horovod/runs/4298078548?check_suite_focus=true#step:163:91